src/CartesianProduct.jl

```diff
@@ -151,6 +151,39 @@ function ∈(x::AbstractVector{<:Real}, cp::CartesianProduct)::Bool
            ∈(view(x, n1+1:length(x)), cp.Y)
 end
 
+"""
+    constraints_list(cp::CartesianProduct{N})::Vector{LinearConstraint{N}} where N<:Real
+
+Return the list of constraints of a (polytopic) Cartesian product.
+
+### Input
+
+- `cp` -- Cartesian product
+
+### Output
+
+A list of constraints.
+"""
+function constraints_list(cp::CartesianProduct{N})::Vector{LinearConstraint{N}} where N<:Real
+    # collect low-dimensional constraints lists
+    clist_low = (constraints_list(cp.X), constraints_list(cp.Y))
+
+    clist = Vector{LinearConstraint{N}}()
+    m = length(clist_low[1]) + length(clist_low[2])
+    sizehint!(clist, m)
+    prev_step = 1
+    # create high-dimensional constraints list
+    for X in clist_low
+        for constr in X
+            new_constr = LinearConstraint(sparsevec(prev_step : (dim(constr) + prev_step-1), constr.a), constr.b)
```

Comment:
The indices are a constant vector for each `X`, so I would pull it out of the inner loop:
```julia
if !isempty(X)
    indices = prev_step : (dim(X[1]) + prev_step-1)
end
for constr in X
    new_constr = LinearConstraint(sparsevec(indices, constr.a), constr.b)
...